openedx/core/djangoapps/credit/partition_schemes.py

```diff
@@ -70,13 +70,20 @@ def get_group_for_user(cls, course_key, user, user_partition):
             # and has either `skipped any ICRV` or `was denied at any ICRV
             # (used all attempts for an ICRV but still denied by the software
             # secure)` or `has submitted/approved verification for given ICRV`
-            return cls.VERIFIED_ALLOW
+            partition_group = cls.VERIFIED_ALLOW
         else:
             # the course content tagged with given 'user_partition' is
             # accessible/visible to the students enrolled as `verified` users
             # and has not yet submitted for the related ICRV
-            return cls.VERIFIED_DENY
+            partition_group = cls.VERIFIED_DENY
+
+        # return matching user partition group if it exists
+        try:
+            return user_partition.get_group(partition_group)
+        except NoSuchUserPartitionGroupError:
+            return None
 
+    # TODO: Remove this method and related tests since it is not used anymore
```

Comment:
@aamir-khan since you are working on test so make sure you update the old test to reflect the changes after you remove `key_for_partition` method.